sdk/core/azure-core-http-jdk-httpclient/pom.xml

```diff
@@ -126,6 +126,21 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
+        <configuration>
+          <compilerArgs>
+            <arg>-Xlint:deprecation</arg>
```

Comment:
Any reason for enabling this only for core modules and not all client libraries?

Reply:
Once this is in I'll be filing issues against SDK area owners to add this, once most libraries have closed their issues I will propagate this into the parent configuration.